ras/uprobe.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM.
+# Author: Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import genio
+from avocado.utils import distro
+from avocado.utils import process
+from avocado.utils import linux_modules
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Uprobe(Test):
+
+    """
+    Test kernel uprobes
+    :avocado: tags=privileged
+    """
+
+    def run_cmd(self, cmd):
+        self.log.info("executing ============== %s =================", cmd)
+        if process.system(cmd, ignore_status=True, sudo=True, shell=True):
+            self.is_fail += 1
+        return
+
+    @staticmethod
+    def run_cmd_out(cmd):
+        return process.system_output(cmd, shell=True, ignore_status=True,
+                                     sudo=True)
+
+    def clear_trace(self):
+        self.debugfs = "/sys/kernel/debug/tracing/"
+        self.tracefs = os.path.join(self.debugfs, "trace")
+        self.uprobes_enable_fs = os.path.join(self.debugfs, "events/uprobes/enable")
+        self.uprobes_event_fs = os.path.join(self.debugfs, "events/uprobes/u_malloc/enable")
+        self.uprobes_events_fs = os.path.join(self.debugfs, "uprobe_events")
```

Comment:
`self.uprobes_events_fs` value over written   from last assignment   i think other assignment not needed